lib/raster/put_row.c

```diff
@@ -514,7 +514,7 @@ static void write_null_bits_compressed(const unsigned char *flags,
     compressed_buf = G_malloc(cmax);
 
     /* compress null bits file with LZ4, see lib/gis/compress.h */
-    nwrite = G_compress(flags, size, compressed_buf, cmax, 3);
+    nwrite = G_compress((char *)flags, size, compressed_buf, cmax, 3);
```

Comment:
This cast from `const unsigned char *` to `char *` fixes compiler warnings but masks the need to synchronize the affected GRASS function definitions. Something for GRASS 8.

Reply:
That should have been `const unsigned char *` to `unsigned char *` cast, `const` being the "only" difference. Updated accordingly.

I guess the G_compress() family of functions could be updated from:
```
int G_compress(unsigned char *, int, unsigned char *, int, int);
```

to
```
int G_compress(const unsigned char *, int, unsigned char *, int, int);
```
where the source is a const pointer. Is this what you were thinking of?

Reply:
Some fns of external libraries used by the `G_compress()` family of fns want `(char *)` as input, e.g. `BZ2_bzBuffToBuffCompress()`. According casts are done in the respective `lib/gis/cmpr*.c` files.

Your updated change makes sense.